File: Codeforces/Contest 12/C_Bono_Tyres_Are_Gone.py
n = int(input())
pile = []
next_tyre = 1
shufflings = 0
for _ in range(2 * n):
    command = list(input().split())
    if len(command) == 2:
        tyre = int(command[1])
        pile.append(tyre)
    else:
        if pile:
            if next_tyre != pile[-1]:
                shufflings += 1
                pile = []
            else:
                pile.pop()
        next_tyre += 1
print(shufflings)

File: Codeforces/Contest 12/C_Bono_Tyres_Are_Gone.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
    int n;
    cin >> n;
    int next_tyre = 1;
    vector<int> pile;
    int shufflings = 0;
    for (int i = 0; i < 2 * n; ++i)
    {
        string command;
        cin >> command;
        if (command == "add")
        {
            int tyre;
            cin >> tyre;
            pile.push_back(tyre);
        }
        else
        {
            if (next_tyre != pile.back())
            {
                sort(pile.rbegin(), pile.rend());
                ++shufflings;
            }
            pile.pop_back();
            ++next_tyre;
        }
    }
    cout << shufflings << endl;
    return 0;
}